layouts/partials/head.html

```diff
@@ -48,28 +39,54 @@
     <![endif]-->
   ` | safeHTML }}
 
-  <!-- Favicon and apple touch icons-->
+  <!-- Favicon and Apple touch icons-->
   <link rel="shortcut icon" href="{{ "img/favicon.ico" | relURL }}" type="image/x-icon" />
   <link rel="apple-touch-icon" href="{{ "img/apple-touch-icon.png" | relURL }}" />
-  <!-- owl carousel css -->
 
+  <!-- owl carousel CSS -->
   <link href="{{ "css/owl.carousel.css" | relURL}}" rel="stylesheet">
   <link href="{{ "css/owl.theme.css" | relURL}}" rel="stylesheet">
 
+  <!-- RSS feed -->
   <link rel="alternate" href="{{ "/index.xml" | absURL }}" type="application/rss+xml" title="{{ .Site.Title }}">
 
-  <!-- Twitter Card meta tags -->
-  <meta name="twitter:card" content="summary_large_image">
-  <meta name="twitter:site" content="@{{ .Site.Params.twitter }}">
-  <meta name="twitter:title" content="{{ if .Title }}{{ .Title }}{{ else }}{{ .Site.Title }}{{ end }}">
-  {{ if .Params.banner }}<meta name="twitter:image" content="{{ .Params.banner | absURL }}">
-  {{ else }}<meta name="twitter:image" content="{{ .Site.Params.twitterImage | absURL }}">
-  {{ end }}<meta name="twitter:description" content="{{ if .Params.description }}{{ .Params.description }}{{ else }}{{ .Site.Params.defaultDescription }}{{ end }}">
-
   <!-- Facebook OpenGraph tags -->
-  <meta property="og:title" content="{{ .Title }}" />
-  <meta property="og:type" content="website" />
-  <meta property="og:url" content="{{ .Permalink }}/" />
-  <meta property="og:image" content="{{ .Site.Params.logo }}" />
+  {{ $is_blog := and (eq .Type "blog") (eq .Kind "page") }}
+  {{ $has_image :=  isset .Params "banner" }}
+  {{ $image := cond $has_image .Params.banner .Site.Params.default_sharing_image }}
+  {{ $image_ext := path.Ext $image }}
+  <meta property="og:locale" content="{{ replace .Site.LanguageCode "-" "_" }}">
+  <meta property="og:site_name" content="{{ .Site.Title }}">
+  <meta property="og:title" content="{{ .Title | markdownify | plainify }}">
+  <meta property="og:type" content="{{ cond $is_blog "article" "website" }}">
+  <meta property="og:url" content="{{ .Permalink }}" />
+  <meta property="og:description" content="{{ $description | markdownify | plainify }}">
+  <meta property="og:image" content="{{ $image | absURL }}">
+  <meta property="og:image:type" content="image/{{ if eq $image_ext ".svg" }}svg+xml{{ else }}{{ trim $image_ext "." }}{{ end }}">
+  {{ with .Params.banner_alt }}<meta property="og:image:alt" content="{{ . | markdownify | plainify }}">{{ end }}
+  {{ $image_local :=  printf "/static/%s" $image}}
+  {{ with (imageConfig $image_local) }}
+    <meta property="og:image:width" content="{{ .Width }}">
+    <meta property="og:image:height" content="{{ .Height }}">
+  {{ end }}
+  {{ with .Lastmod }}<meta property="og:updated_time" content="{{ .Format "2006-01-02T15:04:05Z0700" }}">{{ end }}
```

Comment:
format repeated many times, can you asign a variable and define just once?

Reply:
you really want to introduce a variable for a function parameter??